### PR TITLE
move image verification logic into preppers. replace string->prep map…

### PR DESCRIPTION
**Pull request by nkubala**
… with array

cc @dlorenc

### CODE REVIEW

File: pkg/util/cloud_prepper.go [hunk 1]
```diff
@@ -25,7 +27,24 @@ type CloudPrepper struct {
...
+}
+
+func (p CloudPrepper) GetSource() string {
+	return p.ImagePrepper.Source
+}
+
+func (p CloudPrepper) SupportsImage() bool {
+	pattern := regexp.MustCompile("^.+/.+(:.+){0,1}$")
+	image := p.ImagePrepper.Source
+	if exp := pattern.FindString(image); exp != image || CheckTar(image) {
```
Comment (dlorenc) [inline pkg/util/cloud_prepper.go]: Hmm, why does this support tars?

File: cmd/analyze.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ var analyzeCmd = &cobra.Command{
 	Short: "Analyzes an image: [image]",
 	Long:  `Analyzes an image using the specifed analyzers as indicated via flags (see documentation for available ones).`,
 	Args: func(cmd *cobra.Command, args []string) error {
-		if err := validateArgs(args, checkAnalyzeArgNum, checkArgType); err != nil {
+		if err := validateArgs(args, checkAnalyzeArgNum); err != nil {
```
Comment (dlorenc) [inline cmd/analyze.go]: Now that we don't validate strings up front, how long does it take to fail if we pass an invalid argument?

File: pkg/util/cloud_prepper.go (hunk 1 above)
Comment (nkubala) [reply pkg/util/cloud_prepper.go]: it won't. keeping the functional changes to a minimum in this PR, I fixed this in https://github.com/GoogleCloudPlatform/container-diff/pull/89

File: cmd/analyze.go (hunk 2 above)
Comment (nkubala) [reply cmd/analyze.go]: marginal amount of time more, if any. the validation happens before any processing happens

Comment (dlorenc) [reply cmd/analyze.go]: Would you mind attaching the output of an error case to the PR?

File: pkg/util/cloud_prepper.go (hunk 1 above)
Comment (dlorenc) [reply pkg/util/cloud_prepper.go]: Ah I see, yeah this worked that way before

File: cmd/analyze.go (hunk 2 above)
Comment (nkubala) [reply cmd/analyze.go]: whoops didn't see this, let me generate one

**Comment by nkubala** [comment]
[error.txt](https://github.com/GoogleCloudPlatform/container-diff/files/1316194/error.txt)

**Pull request merged**